src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs

```diff
@@ -840,19 +843,23 @@ private Task<LockFile> ReadLockFile(string projectLockFilePath)
                     continue;
                 }
 
-                var packageInfo = repository.FindPackagesById(library.Name)
-                                            .FirstOrDefault(p => p.Version == library.Version);
-
-                var projectDependency = projectResolver.FindProject(library.Name);
-
-                if (projectDependency != null)
+                if (item.Match.LibraryType.Equals(LibraryTypes.Project))
                 {
+                    var projectDependency = projectResolver.FindProject(library.Name);
                     var projectLibrary = LockFileUtils.CreateLockFileProjectLibrary(project, projectDependency);
 
                     lockFile.ProjectLibraries.Add(projectLibrary);
                 }
-                else if (packageInfo != null)
+                else if (item.Match.LibraryType.Equals(LibraryTypes.Package))
                 {
+                    var packageInfo = repository.FindPackagesById(library.Name)
+                                                .FirstOrDefault(p => p.Version == library.Version);
+
+                    if (packageInfo == null)
+                    {
+                        throw new InvalidOperationException($"Unresolved package: {library.Name}");
```

Comment:
Yes, but it did before. This is really an internal error if this happens. It could only happen if a package was explicitly deleted between the time it was downloaded and the time the lock file is written.